### PR TITLE
Added/updated service wrappers used for making REST calls to security analytics backend plugin

### PR DESCRIPTION
**Pull request by amsiglan**
Signed-off-by: Amardeepsingh Siglani <amardeep7194@gmail.com>

### Description
Added/updated service wrappers used for different workflows like Alerts, Findings, Rules, Notifications, etc.

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/services/RuleService.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    request: OpenSearchDashboardsRequest<{}, GetRulesParams>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<GetRulesResponse> | ResponseError>> => {
+    try {
+      const { prePackaged } = request.query;
+      const params: GetRulesParams = {
+        prePackaged,
+        body: request.body,
+      };
+      console.log(`Making get rules req: ${params}`);
```
Comment (AWSHurneyt) [inline server/services/RuleService.ts:119]: Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?

File: server/services/RuleService.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  /**
+   * Calls backend PUT Rules API.
+   */
+
```
Comment (AWSHurneyt) [inline server/services/RuleService.ts:178]: Not blocking, but there's an extra space here.

File: server/services/RuleService.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        logsource: {
+          product: log_source,
+          service: log_source,
+        },
+        level,
+        falsepositives: false_positives.map((falsePos) => falsePos.value),
+        detection: JSON.parse(detection),
+      };
+
+      console.log(jsonPayload);
```
Comment (AWSHurneyt) [inline server/services/RuleService.ts:219]: Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?

File: server/services/RuleService.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        tags: tags.map((tag) => tag.value),
+        logsource: {
+          product: log_source,
+          service: log_source,
+        },
+        level,
+        falsepositives: false_positives.map((falsePos) => falsePos.value),
+        detection: JSON.parse(detection),
+      };
+      console.log(jsonPayload);
```
Comment (AWSHurneyt) [inline server/services/RuleService.ts:79]: Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?

**Review by AWSHurneyt — APPROVED** [review]

File: server/services/RuleService.ts (hunk 3 above)
Comment (amsiglan) [reply server/services/RuleService.ts:219]: Will be removed

File: server/services/RuleService.ts (hunk 4 above)
Comment (amsiglan) [reply server/services/RuleService.ts:79]: Will be removed

**Pull request merged**